src/ast/dibuilderbpf.cpp

```diff
@@ -0,0 +1,83 @@
+#include "dibuilderbpf.h"
+#include "utils.h"
+
+#include <llvm/IR/Function.h>
+
+namespace bpftrace {
+namespace ast {
+
+DIBuilderBPF::DIBuilderBPF(Module &module) : DIBuilder(module)
+{
+  file = createFile("bpftrace", ".");
```

Comment:
I don't think it creates an actual file. I quickly skimmed the code of `createFile` and IIUC, it should just create a debug info entry which is required to produce correct debug info for a function. We just need to provide a directory and a file name strings to create it, so we give the current directory and ”bpftrace” for the file name. I think that the strings themselves don't mean anything, we could provide anything.

Reply:
I ran bpftrace through strace and can confirm it doesn't create a file at this location. If we print debug output ("-d") then it does attempt to read this file though:
```
openat(AT_FDCWD, "./bpftrace", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

I can make bpftrace read arbitrary data by manually creating a file at the expected location. It's not necessarily a problem but feels a bit dodgy to me.

How about we do `createFile("", "")`? This seems to work and ensures that there can be no real file associated with our `DIFile` object.

Reply:
Done

Reply:
So, unfortunately, we must pass a valid filename, otherwise LLVM IR verification (which we run in codegen tests) complains. I reverted it to the original version.

Reply:
I ran a few more tests and discovered that LLVM dump tries to open the file and either read it (if it's a text file) or mmap it to memory (if it's an ELF). While this shouldn't be harmful, it's certainly an unnecessary operation.

Using `./bpftrace` is therefore not ideal as it will exist when running `./bpftrace -e ...` and will do an unnecessary mmap of the bpftrace binary. So, I changed it to `bpftrace.bpf.o` which is less likely to exist.